src/env.js

```diff
@@ -118,7 +118,7 @@ export class Env {
      * 
      * @overload
      * @param {string} key The environment variable name to retrieve.
-     * @param {string} defaultValue The default value to return if the
+     * @param {string|number} defaultValue The default value to return if the
```

Comment:
@nzakas Given how people actually consume env variables, they're usually strings, numbers (e.g., `PORT`), and booleans (`SOME_FLAG`).  So it's reasonable to restrict to `string | boolean | number` (I didn't think this through very well, and just assumed `number` based on the string coercion happening).

But it's _also_ reasonable to allow a generic type `T` and _not_ coerce to `string`, because:

- `defaultValue` is not inserted into `source`, so it does not have the same restrictions, and
- `String(T)` returns a `string` for any `T`, and
- it is common for an API that allows for a "default value" returns that value verbatim.

I think I'm leaning towards allowing `T`, because it'll create more boilerplate for people who provide a non-string `defaultValue`, who will then need to coerce _back_ to whatever they passed.

Reply:
It makes sense to just leave it as `string` for now, unless we want to break that API.

Reply:
I've reverted this type to `string`

Reply:
> But it's also reasonable to allow a generic type T and not coerce to string

The goal of coercing to a string is to ensure that you're always getting a string, so there is no differentiation between reading from the environment and using your default. Otherwise, you might end up needing to check the type of the returned value. As an example, I wanted to prevent people from needing to do this:

```js
const featureEnabled = env.get("feature-x", false);

if (!featureEnabled || featureEnabled === "false") {
    // disable the feature
}
```

By ensuring there is always a string returned, the user can always perform the same operation to validate the value:

```js
const featureEnabled = env.get("feature-x", false);

if (featureEnabled === "false") {
    // disable the feature
}
```

Reply:
Yes, that's reasonable as well.

Reply:
I'll change it to `string|number|boolean` then.